### ISSUE TITLE
[question] Bactopia Tools - Kraken2 exits silently without completed process

### ISSUE THREAD

**Issue by llk578496** [bactopia/bactopia#288]
Hi @rpetit3 !
We just tried to use the `Kraken2` in `Bactopia Tools`, but it just exit silently without completed process.
May we know do we need to build a `Kraken2` formatted database first?

Thanks a lot!

> (bactopia) gilmansiu3@gilmansiu3-Z490-AORUS-ULTRA:/mnt/data2/Food/REAL$ bactopia --wf kraken2 --bactopia 04_bactopia/Batch001_RM1_CRA_1_Franklin -profile docker
> 2022-03-01 11:54:58:root:INFO - Checking if environment pre-builds are needed, use --verbose to see full details.
> N E X T F L O W  ~  version 21.10.6
> Launching `/home/gilmansiu3/miniconda3/envs/bactopia/share/bactopia-2.0.x/main.nf` [suspicious_cantor] - revision: bd75c553f9
> 
> 
> --------------------------------------------------------------------
>    _                _              _         _              _              
>   | |__   __ _  ___| |_ ___  _ __ (_) __ _  | |_ ___   ___ | |___          
>   | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` | | __/ _ \ / _ \| / __|    
>   | |_) | (_| | (__| || (_) | |_) | | (_| | | || (_) | (_) | \__ \        
>   |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_|  \__\___/ \___/|_|___/ 
>                             |_|                                            
>   bactopia tools kraken2 v2.0.2
>   Taxonomic classification of sequence reads 
> --------------------------------------------------------------------
> Core Nextflow options
>   runName          : suspicious_cantor
>   containerEngine  : docker
>   container        : quay.io/bactopia/bactopia:2.0.2
>   launchDir        : /mnt/data2/Food/REAL
>   workDir          : /mnt/data2/Food/REAL/work
>   projectDir       : /home/gilmansiu3/miniconda3/envs/bactopia/share/bactopia-2.0.x
>   userName         : gilmansiu3
>   profile          : docker
>   configFiles      : /home/gilmansiu3/miniconda3/envs/bactopia/share/bactopia-2.0.x/nextflow.config
> 
> Required Parameters
>   bactopia         : 04_bactopia/Batch001_RM1_CRA_1_Franklin
> 
> Optional Parameters
>   outdir           : 04_bactopia/Batch001_RM1_CRA_1_Franklin
>   run_name         : kraken2
> 
> Nextflow Profile Parameters
>   condadir         : /home/gilmansiu3/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs
>   registry         : quay
>   singularity_cache: /home/gilmansiu3/.bactopia/singularity
> 
> Helpful Parameters
>   wf               : kraken2
> 
> !! Only displaying parameters that differ from the pipeline defaults !!
> --------------------------------------------------------------------
> If you use bactopia for your analysis please cite:
> 
> * Bactopia
>   https://doi.org/10.1128/mSystems.00190-20
> 
> * The nf-core framework
>   https://doi.org/10.1038/s41587-020-0439-x
> 
> * Software dependencies
>   https://bactopia.github.io/acknowledgements/
> --------------------------------------------------------------------
> 
> Each task will use 4 CPUs out of the available 20 CPUs. At most 
> 5 task(s) will be run at a time, this can affect the efficiency 
> of Bactopia. You can use the '-qs' parameter to alter the number of 
> tasks to run at a time (e.g. '-qs 2', means only 2 tasks or a maximum 
> of 8 CPUs will be used at once)
> 
> 
> logs does not appear to be a Bactopia sample, skipping...
> bactopia does not appear to be a Bactopia sample, skipping...
> Found 1 samples to process
> 
> If this looks wrong, now's your chance to back out (CTRL+C 3 times).
> Sleeping for 5 seconds...
> --------------------------------------------------------------------
> [-        ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS -
> [-        ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS -
> 
>     Bactopia Tools: `kraken2 Execution Summary
>     ---------------------------
>     Bactopia Version : 2.0.2
>     Nextflow Version : 21.10.6
>     Command Line     : nextflow run /home/gilmansiu3/miniconda3/envs/bactopia//share/bactopia-2.0.x/main.nf -w /mnt/data2/Food/REAL/work/ --wf kraken2 --bactopia 04_bactopia/Batch001_RM1_CRA_1_Franklin -profile docker
>     Resumed          : false
>     Completed At     : 2022-03-01T11:55:08.141050+08:00
>     Duration         : 7s
>     Success          : true
>     Exit Code        : 0
>     Error Report     : -
>     Launch Dir       : /mnt/data2/Food/REAL
>

**Comment by rpetit3** [comment]
Looking into this now.

You will need to build a kraken database first, I'd recommend checking these out https://benlangmead.github.io/aws-indexes/

**Comment by rpetit3** [comment]
Ok this is fixed: https://github.com/bactopia/bactopia/commit/edafa87905b56ce06a076986730a08430f4c4ba8 

Looks like I forgot to add it to the `bactopia-tools` workflow. I've also made `--kraken2_db` a required argument

I'll get a dev build on my personal anaconda channel here so for you to try (there seems to currently be an issue with Github actions)

**Comment by rpetit3** [comment]
Seems to have fixed itself

```
conda create -n bactopia-dev -c rpetit3 -c conda-forge -c bioconda rpetit3::bactopia
```

This should let you keep moving along, let me know if not!

**Comment by rpetit3** [comment]
Hi @llk578496 

I've released v2.0.3 (https://github.com/bactopia/bactopia/releases/tag/v2.0.3) which includes a fix for this.

I'm going to close for now, feel free to reopen!
Robert
